packit_service/worker/handlers/koji.py

```diff
@@ -264,3 +258,89 @@ def run(self):
+    ):
+        super().__init__(
+            package_config=package_config,
+            job_config=job_config,
+            event=event,
+        )
+        self.koji_build_event: KojiBuildEvent = KojiBuildEvent.from_event_dict(event)
+        self._db_trigger: Optional[AbstractTriggerDbType] = None
+        self._build: Optional[KojiBuildModel] = None
+
+    @property
+    def build(self) -> Optional[KojiBuildModel]:
+        if not self._build:
+            self._build = KojiBuildModel.get_by_build_id(
+                build_id=str(self.koji_build_event.build_id)
+            )
+        return self._build
+
+    @property
+    def db_trigger(self) -> Optional[AbstractTriggerDbType]:
+        if not self._db_trigger and self.build:
+            self._db_trigger = self.build.get_trigger_object()
+        return self._db_trigger
+
+    def run(self):
+        build = KojiBuildModel.get_by_build_id(build_id=self.koji_build_event.build_id)
+
+        if not build:
+            msg = f"Koji build {self.koji_build_event.build_id} not found in the database."
+            logger.debug(msg)
+            return TaskResults(success=True, details={"msg": msg})
+
+        logger.debug(
+            f"Build on {build.target} in koji changed state "
+            f"from {self.koji_build_event.old_state} to {self.koji_build_event.state}."
+        )
+
+        new_commit_status = {
+            KojiBuildState.building: BaseCommitStatus.running,
+            KojiBuildState.complete: BaseCommitStatus.success,
+            KojiBuildState.deleted: BaseCommitStatus.error,
+            KojiBuildState.failed: BaseCommitStatus.failure,
+            KojiBuildState.canceled: BaseCommitStatus.error,
+        }.get(self.koji_build_event.state)
+
+        if (
+            new_commit_status
+            and build.status
+            and build.status != KojiBuildState.building
+        ):
+            logger.warning(
+                f"We should not overwrite the final state {build.status} "
+                f"to {self.koji_build_event.state}. "
+                f"Not updating the status."
+            )
+        elif new_commit_status:
+            build.set_status(new_commit_status.value)
+        else:
+            logger.debug(
+                f"We don't react to this koji build state change: {self.koji_task_event.state}"
+            )
+
+        if not build.web_url:
+            build.set_web_url(
+                KojiBuildEvent.get_koji_rpm_build_web_url(
+                    rpm_build_task_id=self.koji_build_event.rpm_build_task_id,
+                    koji_web_url=self.service_config.koji_web_url,
+                )
+            )
+        # TODO: update logs URL (the access via task number dos not work for non-scratch builds)
+
+        msg = (
+            f"Build on {build.target} in koji changed state "
+            f"from {self.koji_build_event.old_state} to {self.koji_build_event.state}."
+        )
```

Comment:
We already have this at the beginning of the method
```suggestion
```

Reply:
Wait, that's for the task result, not for logging. I'll revert that change.